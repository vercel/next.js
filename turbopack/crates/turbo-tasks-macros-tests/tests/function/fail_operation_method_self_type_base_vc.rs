#![allow(dead_code)]
#![feature(arbitrary_self_types)]
#![feature(arbitrary_self_types_pointers)]

use turbo_tasks::Vc;

#[turbo_tasks::value]
struct Foobar;

#[turbo_tasks::value_impl]
impl Foobar {
    #[turbo_tasks::function(operation)]
    fn arbitrary_self_type_base_vc(self: Vc<Self>) -> Vc<()> {
        Vc::cell(())
    }
}

fn main() {}
