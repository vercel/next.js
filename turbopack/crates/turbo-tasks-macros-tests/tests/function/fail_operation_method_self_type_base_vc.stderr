error: methods taking `self` are not supported with `operation`
  --> tests/function/fail_operation_method_self_type_base_vc.rs:13:36
   |
13 |     fn arbitrary_self_type_base_vc(self: Vc<Self>) -> Vc<()> {
   |                                    ^^^^^^^^^^^^^^

error[E0277]: the trait bound `Vc<Foobar>: NonLocalValue` is not satisfied
  --> tests/function/fail_operation_method_self_type_base_vc.rs:13:42
   |
13 |     fn arbitrary_self_type_base_vc(self: Vc<Self>) -> Vc<()> {
   |                                          ^^^^^^^^ the trait `NonLocalValue` is not implemented for `Vc<Foobar>`
   |
   = help: the following other types implement trait `NonLocalValue`:
             &T
             &mut T
             ()
             (A, Z, Y, X, W, V, U, T)
             (B, A, Z, Y, X, W, V, U, T)
             (C, B, A, Z, Y, X, W, V, U, T)
             (D, C, B, A, Z, Y, X, W, V, U, T)
             (E, D, C, B, A, Z, Y, X, W, V, U, T)
           and $N others
note: required by a bound in `assert_argument_is_non_local_value`
  --> $WORKSPACE/turbopack/crates/turbo-tasks/src/macro_helpers.rs
   |
   | pub fn assert_argument_is_non_local_value<Argument: NonLocalValue>() {}
   |                                                     ^^^^^^^^^^^^^ required by this bound in `assert_argument_is_non_local_value`
